src/dtos/fileimport-dto.ts

```diff
@@ -0,0 +1,42 @@
+import { FileImport } from '../entities/file-import';
+import { Source } from '../entities/source';
+
+import { SourceDTO } from './source-dto';
+
+export class ImportDTO {
+    id: string;
+    revision_id: string;
+    mime_type: string;
+    filename: string;
+    hash: string;
+    uploaded_at: string;
+    type: string;
+    location: string;
+    sources?: SourceDTO[];
+
+    static async fromImport(importEntity: FileImport): Promise<ImportDTO> {
+        const dto = new ImportDTO();
+        dto.id = importEntity.id;
+        const revision = await importEntity.revision;
+        dto.revision_id = revision.id;
+        dto.mime_type = importEntity.mimeType;
+        dto.filename = importEntity.filename;
+        dto.hash = importEntity.hash;
+        dto.uploaded_at = importEntity.uploadedAt?.toISOString() || '';
```

Comment:
If this is optional then you don't need the empty string fallback, undefined is fine.

Reply:
Fixed this.